app/src/main/java/io/pslab/communication/CommunicationHandler.java

```diff
@@ -57,18 +59,24 @@ public CommunicationHandler(UsbManager usbManager) {
         mWriteBuffer = new byte[DEFAULT_WRITE_BUFFER_SIZE];
     }
 
-    public void open() throws IOException {
+    public void open(int baudRate) throws IOException {
         if (!device_found) {
             throw new IOException("Device not Connected");
         }
         mConnection = mUsbManager.openDevice(mUsbDevice);
         if (mConnection == null) {
             throw new IOException("Could not open device.");
         }
+        if (mUsbDevice.getProductId() == PSLAB_PRODUCT_ID_V6 && mUsbDevice.getVendorId() == PSLAB_VENDOR_ID_V6) {
+            PSLAB_VERSION = 6;
+        }
+        else {
```

Comment:
Similar to the question as in PowerSourceActivity: Should we explicitly check for the PSLab v5 product ID and vendor ID and throw an error in case of unexpected values in an extra `else` after this block?